javascript/ql/src/semmle/javascript/dataflow/TypeInference.qll

```diff
@@ -47,6 +47,12 @@ class AnalyzedNode extends DataFlow::Node {
    */
   AnalyzedNode localFlowPred() { result = getAPredecessor() }
 
+  /**
+   * Gets another data flow node whose value flows into this node in a global step
+   * (this is, involving global variables).
```

Comment:
Perhaps it should be clarified that this is an extension point that should be used with care.

Reply:
Marking it as internal for now. Overriding it is no less dangerous than overriding `localFlowPred`, not that people tend to subclass `AnalyzedNode` in the first place.